test/compound/TestLens.t.sol

```diff
@@ -1719,4 +1721,40 @@ contract TestLens is TestSetup {
         morpho.setIsLiquidateBorrowPaused(cDai, true);
         assertTrue(lens.getMarketPauseStatus(cDai).isLiquidateBorrowPaused);
     }
+
+    function testPoolIndexGrowthInsideBlock() public {
+        supplier1.approve(dai, type(uint256).max);
+        supplier1.supply(cDai, 1 ether);
+
+        uint256 poolBorrowIndexBefore = lens.getIndexes(cDai, true).poolSupplyIndex;
+
+        vm.prank(address(supplier1));
+        ERC20(dai).transfer(cDai, 10_000 ether);
+
+        supplier1.supply(cDai, 1);
+
+        uint256 poolSupplyIndexAfter = lens.getIndexes(cDai, true).poolSupplyIndex;
+
+        assertGt(poolSupplyIndexAfter, poolBorrowIndexBefore);
+    }
+
+    function testP2PIndexGrowthInsideBlock() public {
+        borrower1.approve(dai, type(uint256).max);
+        borrower1.supply(cDai, 1 ether);
+        borrower1.borrow(cDai, 0.5 ether);
+        setDefaultMaxGasForMatchingHelper(0, 0, 0, 0);
+        // Bypass the borrow repay in the same block by overwritting the storage slot lastBorrowBlock[borrower1].
+        hevm.store(address(morpho), keccak256(abi.encode(address(borrower1), 178)), 0);
+        // Create delta.
+        borrower1.repay(cDai, type(uint256).max);
+
+        uint256 p2pSupplyIndexBefore = lens.getCurrentP2PSupplyIndex(cDai);
+
+        vm.prank(address(supplier1));
+        ERC20(dai).transfer(cDai, 10_000 ether);
+
```

Comment:
It's because the lens is looking for the pool indexes on the pool, so we will notice the change. In the other tests we want to notice the changes on Morpho, so we have to wait for one block and interact with Morpho